content/blog/2022-03-31-training-and-saving-models-on-a-dedicated-runner.md

```diff
@@ -0,0 +1,276 @@
+our CI/CD to provision a runner and run the script. We define our workflow in
+`cml.yaml` and save it in the `.github/workflows` directory. This way GitHub
+will automatically go through the workflow whenever it is triggered. In this
+case the triggers are a manual run and the daily schedule.
+
+<admon type="info">
+The name of the workflow doesn’t matter, as long as it’s a <code>.yaml</code> and
+located in the <code>.github/workflows</code> directory. You can have multiple workflows
+in there as well. You can learn more in the
+<a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions">documentation</a>
+here.
+</admon>
+
+```yaml
+name: CML
+on: # Here we use two triggers: manually and daily at 08:00
+  workflow_dispatch:
+  schedule:
+    - cron: '0 8 * * *'
+jobs:
+  deploy-runner:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: iterative/setup-cml@v1
+      - uses: actions/checkout@v2
+      - name: Deploy runner on EC2
+        env:
+          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          cml runner \
+              --cloud=aws \
+              --cloud-region=eu-west \
+              --cloud-type=t2.micro \
+              --single
+  train-model:
+    needs: deploy-runner
+    runs-on: [self-hosted]
+    timeout-minutes: 120 # 2h
+    container:
+      image: docker://iterativeai/cml:0-dvc2-base1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Train model
+        env:
+          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+        run: |
+          cml ci
+          pip install -r requirements.txt
+          python train.py
+```
+
+<admon type="warn">
+In this example we are using a <code>t2.micro</code> AWS EC2 instance. At the time of writing this is included in the AWS free tier. Make sure that you qualify for this free usage to prevent unexpected spending. When you specify a bulkier <code>cloud-type</code>, your expenses will rise.
+</admon>
+
+The workflow we defined first provisions a runner on AWS and then uses that
+runner to train the model. After completing the training job, CML automatically
+terminates the runner to prevent you from incurring further costs. Once the
+runner is terminated, however, the model is lost along with it. Let's see how we
+can save our model in the next step!
+
+# Export the model to our Git repository
+
+CML allows us to export the model from our runner to our Git repository. Let's
+extend the training stage of our workflow by pushing `random_forest.joblib` to a
+new experiment branch and creating a merge request.
+
+`cml pr` is the command that specifies which files should be included in the
+merge request. The commands after that are used to generate a report in the
+merge request that displays the confusion matrix and calculated metrics.
```

Comment:
```suggestion
pull request. The commands after that are used to generate a report in the
pull request that displays the confusion matrix and calculated metrics.
```

since we're using GH not GL?